addons/binding/org.openhab.binding.camera/src/main/java/org/openhab/binding/camera/handler/CameraHandler.java

```diff
@@ -0,0 +1,195 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link CameraHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Thomas Hartwig - Initial contribution
+ */
+public class CameraHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(CameraHandler.class);
+    private final AtomicBoolean refreshInProgress = new AtomicBoolean(false);
+    private final AtomicBoolean initialized = new AtomicBoolean(false);
+    private final ExecutorService serviceCached = Executors.newCachedThreadPool();
+    private String urlSnapshot;
+    private String urlUsername = "";
+    private String urlPassword = "";
+
+    public CameraHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(CHANNEL_IMAGE)) {
+            if (command.toString().equals("REFRESH")) {
+                refreshData();
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        if (logger.isDebugEnabled()) {
+            logger.debug("Initialize thing: {}::{}", getThing().getLabel(), getThing().getUID());
+        }
+        Object paramUrl = getConfig().get("urlSnapshot");
+        urlSnapshot = String.valueOf(paramUrl);
+        Object paramUsername = getConfig().get("urlUsername");
+        if (paramUsername != null) {
+            urlUsername = String.valueOf(paramUsername);
+        }
+        Object paramPassword = getConfig().get("urlPassword");
+        if (paramPassword != null) {
+            urlPassword = String.valueOf(paramPassword);
+        }
+        try {
+            Object param = getConfig().get("poll");
+            CONFIGURATION_POLLTIME_MS = (int) (Double.parseDouble(String.valueOf(param)) * 1000);
+            logger.debug("Schedule update at fixed rate {} ms.", CONFIGURATION_POLLTIME_MS);
+        } catch (Exception e1) {
+            logger.warn("could not read poll time", e1);
+        }
+        if (initialized.compareAndSet(false, true)) {
+            WeakReference<CameraHandler> weakReference = new WeakReference<>(this);
+            serviceCached.submit(new Callable<Void>() {
+                @Override
+                public Void call() throws Exception {
+                    while (weakReference.get() != null) {
+                        try {
+                            refreshData();
+                        } catch (Exception e) {
+                            logger.error("error in refresh", e);
+                        }
+                        Thread.sleep(Math.max(10, CONFIGURATION_POLLTIME_MS));
```

Comment:
This is not the way, please use a scheduler, which is available from the parent class, you can then schedule it to run every polltime. See examples in other bindings. Questions, let me know!

Reply:
I have a problem using a scheduler here. As I have learned a binding might be reconfigured and initialized during a reconfiguration process without being reinstantiated*. The reconfiguration can affect my polling. This is not possible when using the parent scheduler.scheduleAtFixedRate. Of course I can submit my callable instead to my service to the parent scheduler with my poll time sleep which was reread from configuration. Is this better?

*) I am not right sure if this really is true. If it is the case many other bindings might be wrong and use the scheduler wrong then. This is why I used the initializer atomic here.